test/integration/component/maint/test_capacity_host_delete.py

```diff
@@ -0,0 +1,195 @@
+#    same as above host is not disabled
+#
+
+def update_host(apiclient, state, host_id):
+    """
+    Function to Enable/Disable Host
+    """
+    host_status = Host.update(
+        apiclient,
+        id=host_id,
+        allocationstate=state
+    )
+    return host_status.resourcestate
+
+
+def check_db(self, host_state):
+    """
+    Function to check capacity_state in op_host_capacity table
+    """
+    capacity_state = self.dbclient.execute(
+        "select capacity_state from op_host_capacity where host_id='%s' and capacity_type in (0,1) order by capacity_type asc;" %
+        self.host_db_id[0][0])
+    self.assertEqual(
+        capacity_state[0][0],
+        host_state +
+        "d",
+        "Invalid db query response for capacity_state %s" %
+        capacity_state[0][0])
+    self.assertEqual(
+        capacity_state[1][0],
+        host_state +
+        "d",
+        "Invalid db query response for capacity_state %s" %
+        capacity_state[1][0])
+
+    capacity_state = self.dbclient.execute(
+        "select capacity_state from op_host_capacity where host_id='%s' and capacity_type = 3 order by capacity_type asc;" %
+        self.host_db_id[0][0])
+    self.assertNotEqual(
+        capacity_state[0][0],
+        host_state +
+        "d",
+        "Invalid db query response for capacity_state %s" %
+        capacity_state[0][0])
+    return capacity_state[0][0]
+
+
+class TestHosts(cloudstackTestCase):
+
+    """
+    Testing Hosts
+    """
+    @classmethod
+    def setUpClass(cls):
+        cls.testClient = super(TestHosts, cls).getClsTestClient()
+        cls.testdata = cls.testClient.getParsedTestDataConfig()
+        cls.apiclient = cls.testClient.getApiClient()
+        cls.dbclient = cls.testClient.getDbConnection()
+        cls._cleanup = []
+
+        # get zone, domain etc
+        cls.zone = Zone(get_zone(cls.apiclient, cls.testClient.getZoneForTests()).__dict__)
+        cls.domain = get_domain(cls.apiclient)
+        cls.pod = get_pod(cls.apiclient, cls.zone.id)
+
+        cls.logger = logging.getLogger('TestHosts')
+        cls.stream_handler = logging.StreamHandler()
+        cls.logger.setLevel(logging.DEBUG)
+        cls.logger.addHandler(cls.stream_handler)
+
+        # list hosts
+        hosts = list_hosts(cls.apiclient, type="Routing")
+        i = 0
+        while (i < len(hosts)):
+            host_id = hosts[i].id
+            cls.logger.debug("Trying host id : %s" % host_id)
+            host_db_id = cls.dbclient.execute(
+                "select id from host where uuid='%s';" %
+                host_id)
+
+            if host_db_id and host_db_id[0]:
+                cls.logger.debug("found host db id : %s" % host_db_id)
+                storage_pool_db_id = cls.dbclient.execute(
+                    "select id from storage_pool where id='%s' and removed is null;" %
+                    host_db_id[0][0])
+
+            if storage_pool_db_id and storage_pool_db_id[0]:
```

Comment:
If the condition `if host_db_id and host_db_id[0]:` evaluates to false, then `storage_pool_db_id` will not be defined and I think this will cause a problem because the variable will not be defined.

For example:
```
NameError: name 'storage_pool_db_id' is not defined
```
You should be making sure the variable is defined outside the conditional otherwise you can run into this...